packages/runner/src/storage/cache.ts

```diff
@@ -1242,15 +1242,21 @@ export class Replica {
   // Session Identity. Skips if ACL already exists, or if
   // there is no user or space identity.
   private async requestACLInit() {
-    if (!this.spaceIdentity) {
-      return;
-    }
-    const aclKey = this.spaceIdentity.did();
     const userIdentity = (this.remote as any).as as Signer | undefined;
     if (!userIdentity) {
       return;
     }
 
+    // Request ACL in two cases:
+    // 1. When spaceIdentity exists (delegated space)
+    // 2. When user's identity equals the space DID (home space)
+    const isHomeSpace = userIdentity.did() === this.space;
+    if (!this.spaceIdentity && !isHomeSpace) {
+      return;
+    }
```

Comment:
More documentation of reasoning needed? @jsantell?

Reply:
This looks good :+1: